### ISSUE TITLE
Update a11y contributor guidelines

### ISSUE THREAD

**Issue by LHinson** [patternfly/patternfly-next#528]
The contributor guidelines should be updated to reflect the latest recommendations for tools. 

As part of the [tooling added to check for a11y violations](https://github.com/patternfly/patternfly-next/pull/517), the contributor guidelines should include some instruction on how to set that up, as well as expectations on what to do when violations are found. During this initial exploration phase, some of the ways we handle violations could be one of the following:

- remove the rule if it isn't relevant for the type of example (e.g. components aren't going to have landmark roles, but page layouts would)
- fix the violation
- open an issue to further investigate the violation if it isn't obvious how to fix or move forward with it.

**Comment by jgiardino** [comment]
I met with @seanforyou23 to draft a section on how to use the a11y audit tool and address violations. I included this draft below, but we're not sure where exactly it should go. Some possible places are the [contributing guide](https://pf-next.com/contribution) or the [readme](https://github.com/patternfly/patternfly-next ), but we'd love some feedback on that (@mcarrano @stacymcauliffe @dgutride @LHinson). 

We're expecting that issue #541 will still be open when this information is added to our workspace/repo.

As part of this update, I also expect to include a numbered list item to the [Testing section of the A11y Guide](https://pf-next.com/accessibility-guide#testing) that will link to this information.


## Testing for Accessibility

PatternFly uses [aXe: The Accessibility Engine](https://www.deque.com/axe/) to check for accessibility violations. Our goal is to meet WCAG 2.0 AA requirements, as noted in our [Accessibility Guide](https://pf-next.com/accessibility-guide).

### How to Perform an Accessibility Audit with aXe
aXe is available as either a browser extension or npm script.

To run the script, run the following commands:
```
npm run dev
npm run a11y (in a separate terminal)
```
The tool is configured to return WCAG 2.0 AA violations for the full page renderings of all components, layouts, utilities, and demos. The tool will provide feedback about what the violation is and a link to documentation about how to address the violation.

The same tool is also available as a browser extension for [Chrome](https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd) and [Firefox](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/). 

### Fixing Violations

Ignore the violations that aren’t related to your contribution. _We’re still working on existing violations at this time._

Fix violations related to your contribution.

If there are violations that are not obvious how to fix, then create an issue with information about the violation that was found. For example, some violations might require further discussion before they can be addressed. And some violations may not be valid and require changes to the workspace or tooling to stop flagging the violation.

**Comment by stacymcauliffe** [comment]
@jgiardino - are we expecting users of _and_ contributors to PF will need to use this tool? Or only contributors? If it's only for contributors, then I think we'd want to include it in the contributing guide and make sure we're creating a clear separation between the content in the readme and the content in the contributing guide. 

If it's for both types of users, then we'll probably want it on both places and we'll want to make it explicit how we expect them to use the tool for each different use case. 

This relates a bit to an [issue](https://github.com/patternfly/patternfly/issues/978) I need to pick back up that looks at what content is appropriate for a readme vs contributing.

**Comment by jgiardino** [comment]
@stacymcauliffe Thanks for the feedback! That's a great distinction. This draft is for contributors. Right now, both the [contributing guide](https://pf-next.com/contribution) and the [readme](https://github.com/patternfly/patternfly-next ) seem to be focused on contributors, but I definitely agree that we need a clear separation between contents intended for contributors and contents intended for consumers.

On that topic, should we provide a section for consumers about getting this tool set up in their own project? I'm kinda thinking we recommend the tool, and provide a link, but have the consumer review the tool documentation about how to set it up in their project. I can add that to the [Accessibility Guide](https://pf-next.com/accessibility-guide) as part of this. I included a draft for that in my original comment above.
